### PR TITLE
fix problem with callback params

### PR DESCRIPTION
**Pull request by benjamin-work**
Addresses #175 and #148\, supersedes #178\.

Additionally, fixes several bugs related to `get_params` and `set_params` that were discovered during the changes.

### CODE REVIEW

File: skorch/net.py [hunk 1]
```diff
@@ -352,13 +353,30 @@ def initialize_callbacks(self):
...
                                  "once.".format(name))
             names_seen.add(name)
 
+            # check if callback itself is changed
+            param_callback = getattr(self, 'callbacks__' + name, Dummy)
+            if param_callback is not Dummy:  # callback itself was set
+                cb = param_callback
+
+            # check for callback params
+
```
Comment (ottonemo) [inline skorch/net.py]: Weird newline

File: skorch/net.py [hunk 2]
```diff
@@ -352,13 +353,30 @@ def initialize_callbacks(self):
...
+            pass
+
         for name, cb in self._yield_callbacks():
             if name in names_seen:
                 raise ValueError("The callback name '{}' appears more than "
                                  "once.".format(name))
             names_seen.add(name)
 
+            # check if callback itself is changed
+            param_callback = getattr(self, 'callbacks__' + name, Dummy)
```
Comment (ottonemo) [inline skorch/net.py]: Why use a Dummy class and not `None`? (Maybe shortly document why)

File: skorch/net.py [hunk 3]
```diff
@@ -1055,8 +1073,27 @@ def _get_params_for_optimizer(self, prefix, named_parameters):
...
+                continue
+            for subkey, subval in val.get_params().items():
+                subname = name + '__' + subkey
+                params[subname] = subval
+        return params
+
     def get_params(self, deep=True, **kwargs):
-        return BaseEstimator.get_params(self, deep=deep, **kwargs)
+        params = BaseEstimator.get_params(self, deep=deep, **kwargs)
+        params_cb = self._get_params_callbacks(deep=deep)
```
Comment (ottonemo) [inline skorch/net.py]: Why are callbacks special (short comment)

File: skorch/net.py [hunk 4]
```diff
@@ -1055,8 +1073,27 @@ def _get_params_for_optimizer(self, prefix, named_parameters):
     def _get_param_names(self):
         return self.__dict__.keys()
 
+    def _get_params_callbacks(self, deep=True):
```
Comment (ottonemo) [inline skorch/net.py]: Which problem does this solve and how? (short documentation)

Comment (benjamin-work) [reply skorch/net.py]: Added a comment

**Comment by benjamin-work** [comment]
@ottonemo pls review again

**Pull request merged**